mllib/src/main/scala/org/apache/spark/mllib/fpm/LocalPrefixSpan.scala

```diff
@@ -86,16 +77,18 @@ private[fpm] object LocalPrefixSpan extends Logging with Serializable {
    * Generates frequent items by filtering the input data using minimal count level.
    * @param minCount the minimum count for an item to be frequent
    * @param database database of sequences
-   * @return item and count pairs
+   * @return freq item to count map
    */
   private def getFreqItemAndCounts(
       minCount: Long,
-      database: Array[Array[Int]]): Iterable[(Int, Long)] = {
-    database.flatMap(_.distinct)
```

Comment:
This will create an `Array[Array[Int]]` object, while we only need to scan through it one by one.